From 87fc396d28eb08482cfa2fd8b0dcf031c2b4cffd Mon Sep 17 00:00:00 2001
From: Peter Shen <peter.shen@amd.com>
Date: Tue, 2 Jun 2026 19:30:07 +0800
Subject: [PATCH] ARM: dts: aspeed: anacapa: evt2: add shunt resistor values
 for HSC monitors

Add shunt resistor configuration for LTC4287 hot-swap controllers on
Anacapa EVT2 platform.

The resistor values are required for accurate current and power
measurement by the driver. These settings reflect the actual BOM used
on EVT2, where different sense resistor values (150 uOhm and 250 uOhm)
are populated across rails.

Without this configuration, the reported readings would be incorrect.

LORE: https://lore.kernel.org/all/20260602-anacapa-devlop-phase-devicetree-v3-9-7c93c5df8d9b@gmail.com/

Signed-off-by: Peter Shen <peter.shen@amd.com>
---
 .../aspeed-bmc-facebook-anacapa-evt2.dts      | 37 +++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-anacapa-evt2.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-anacapa-evt2.dts
index ce501ccfb..27d465f34 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-anacapa-evt2.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-anacapa-evt2.dts
@@ -545,6 +545,27 @@ gpio@24 {
 					"", "";
 			};
 
+			power-monitor@42 {
+				compatible = "lltc,ltc4287";
+				reg = <0x42>;
+				shunt-resistor-micro-ohms = <150>;
+			};
+			power-monitor@43 {
+				compatible = "lltc,ltc4287";
+				reg = <0x43>;
+				shunt-resistor-micro-ohms = <150>;
+			};
+			power-monitor@44 {
+				compatible = "lltc,ltc4287";
+				reg = <0x44>;
+				shunt-resistor-micro-ohms = <250>;
+			};
+			power-monitor@45 {
+				compatible = "lltc,ltc4287";
+				reg = <0x45>;
+				shunt-resistor-micro-ohms = <250>;
+			};
+
 			// R-PDB FRU
 			eeprom@50 {
 				compatible = "atmel,24c128";
@@ -590,6 +611,22 @@ gpio@24 {
 					"","";
 			};
 
+			power-monitor@40 {
+				compatible = "lltc,ltc4287";
+				reg = <0x40>;
+				shunt-resistor-micro-ohms = <150>;
+			};
+			power-monitor@41 {
+				compatible = "lltc,ltc4287";
+				reg = <0x41>;
+				shunt-resistor-micro-ohms = <150>;
+			};
+			power-monitor@45 {
+				compatible = "lltc,ltc4287";
+				reg = <0x45>;
+				shunt-resistor-micro-ohms = <250>;
+			};
+
 			// L-PDB FRU
 			eeprom@50 {
 				compatible = "atmel,24c128";
